crowbar_framework/lib/crowbar/upgrade_status.rb

```diff
@@ -130,8 +130,14 @@ def end_step(success = true, errors = {})
       end
     end
 
+    # Check if given step is running.
+    # Without argument, check if any step is running
     def running?(step_name = nil)
-      step = progress[:steps][step_name || current_step]
+      if step_name.nil?
+        return progress[:steps].select { |_, s| s[:status] == :running }.any?
```

Comment:
@Itxaka this seems to be a common practice in ruby. I hit it already a couple of times.